yarn-project/types/src/abi/contract_artifact.ts

```diff
@@ -256,7 +256,23 @@ function getStorageLayout(input: NoirCompiledContract) {
 function getNoteTypes(input: NoirCompiledContract) {
   type t = {
     kind: string;
-    fields: [{ kind: string; sign: boolean; value: string }, { kind: string; value: string }];
+    fields: [
```

Comment:
This type probably shouldn't be repeated (there's `AbiValue` and `BasicValue` in this same package)

Reply:
It's an ugly hack but without it it does not compile. Without it it does not compile because TS complains when you don't have a guarantee of a given property being on the type. The `AbiValue` type is problematic.

<img width="1538" alt="image" src="https://github.com/user-attachments/assets/b05f7647-a776-4396-984f-3266b09b3245">

Reply:
Addressed in later commits.